src/packages/emmett-mongodb/src/eventStore/mongoDBEventStore.ts

```diff
@@ -277,14 +315,57 @@ export class MongoDBEventStore implements EventStore<MongoDBReadEventMetadata> {
         currentStreamVersion === MongoDBEventStoreDefaultStreamVersion,
     };
   }
+
+  close(): Promise<void> {
+    if (this.isClosed) return Promise.resolve();
+
+    this.isClosed = true;
+    if (!this.shouldManageClientLifetime) return Promise.resolve();
```

Comment:
Probably should add some docs either on `close` or on the `client` option to make sure the user knows that we only attempt to close the `client` if we are managing it. Looking at the code I can see this and understand why but I can imagine a scenario where a user attempts to close the event store but they had passed a `client`, and even though `this.isClosed` is true, the client is still open and calling this function has no effect on it.

Reply:
We might also want to add some checks for `isClosed` in `collectionFor` or the individual functions for interacting with the event store. I _think_ if the client is actually closed it'll throw an error when you try to use it and it's been closed, but in the above example, you could technically call close on the eventstore even if you passed your own client and it essentially does nothing.

I might be missing something obvious in why this was intended, just my thoughts.

Reply:
> Probably should add some docs either on close or on the client option to make sure the user knows that we only attempt to close the client if we are managing it. 

Yes, makes sense. Besides docs, I could also make some typescript Voodoo with type interference and expose the `close` method only if the user passed the client through options (so essentially, we'd have different types for managed event store and "ambient" one). But I'm not sure if it'd be more intuitive or not. Currently calling close is a safe default for both options, as it shouldn't cause any side-effects. Thoughts? :thinking: 

> We might also want to add some checks for isClosed in collectionFor or the individual functions for interacting with the event store.

I was thinking on it, but then if someone closed the event store then it'll get the same MongoDB errors as they'd get from a regular usage. But it may be worth indeed wrapping it with some helper and more user-friendly Emmett errors. Thoughts? :thinking:

Reply:
Yeah that makes sense, I think that's fine. Some TS magic to _not_ expose `close` if client is provided would probably be enough.

Reply:
Added issue for that: https://github.com/event-driven-io/emmett/issues/143